### PR TITLE
ref: Generate joint dev & prod pinned requirements file

### PR DESCRIPTION
**Pull request by armenzg**
This prevents conflicts of package versions between the two.

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -20,8 +20,7 @@ dotenv_if_exists .env
 if [ ! -d venv ]; then
     python3 -m venv venv
     venv/bin/pip install wheel
-    venv/bin/pip install -r requirements.txt
-    venv/bin/pip install -r requirements.dev.txt
+    venv/bin/pip install -r requirements.dev-prod.txt
```
Comment (armenzg) [inline .envrc]: The produced requirements is generated based on the prod and input files.

File: README.md [hunk 2]
```diff
@@ -29,8 +29,8 @@ To test against another repo you can use the env variable `GETSENTRY_REPO` in or
 If you're changing the requirements, after having activated the venv and installed the dependencies run the following:
 
 ```shell
-python -m piptools compile > requirements.txt
-python -m piptools compile requirements.dev.in > requirements.dev.txt
+python -m piptools compile requirements.in -o requirements.txt
```
Comment (armenzg) [inline README.md]: The changes on this line is only to make it more similar to the next one.

File: README.md [hunk 3]
```diff
@@ -29,8 +29,8 @@ To test against another repo you can use the env variable `GETSENTRY_REPO` in or
...
+python -m piptools compile requirements.in requirements.dev.in -o requirements.dev-prod.txt
```
Comment (armenzg) [inline README.md]: Taking the prod and dev files as input at the same time.

File: .envrc [hunk 4]
```diff
@@ -20,7 +20,6 @@ dotenv_if_exists .env
 if [ ! -d venv ]; then
     python3 -m venv venv
     venv/bin/pip install wheel
-    venv/bin/pip install -r requirements.txt
     venv/bin/pip install -r requirements.dev.txt
```
Comment (armenzg) [inline .envrc:23]: In my previous commit, I had renamed this file to be `requirements.dev-prod.txt`, however, Github not understand the move. I will follow up.

File: requirements.dev.txt [hunk 5]
```diff
@@ -2,35 +2,79 @@
...
+#    pip-compile --output-file=requirements.dev.txt requirements.dev.in requirements.in
 #
 appdirs==1.4.4
     # via black
 attrs==21.4.0
     # via pytest
 black==21.7b0
     # via -r requirements.dev.in
+blinker==1.4
+    # via sentry-sdk
```
Comment (armenzg) [inline requirements.dev.txt]: I will remove the annotations on a following PR. I wanted to reduce the number of changes on the PR.

File: requirements.dev.txt [hunk 6]
```diff
@@ -2,36 +2,64 @@
...
 # To update, run:
 #
-#    pip-compile --no-annotate --output-file=requirements.dev.txt requirements.dev.in
+#    pip-compile --no-annotate --output-file=requirements.dev.txt requirements.dev.in requirements.in
 #
 appdirs==1.4.4
 attrs==21.4.0
 black==21.7b0
+blinker==1.4
+cachetools==4.2.4
```
Comment (armenzg) [inline requirements.dev.txt:11]: The diff of this file should mainly have the addition of the pinned versions from production.

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**